microhapulator/api.py

```diff
@@ -684,13 +684,16 @@ def plot_haplotype_calls(result, outdir, sample=None, plot_marker_name=True, ign
     plt.switch_backend(backend)
 
 
-def get_reads_in_marker_loci(fullref_bam_file, mhindex):
-    fullref_bam = pysam.AlignmentFile(fullref_bam_file, "rb")
+def get_reads_in_marker_loci(marker_bam, fullrefr_bam, mhindex):
+    fullref_bam = pysam.AlignmentFile(fullrefr_bam, "rb")
     reads_to_markers = defaultdict(list)
     for locus in mhindex.loci.values():
         start, end = locus.span(chrom=True)
+        baseline = set()
+        for read in marker_bam.fetch(locus.id):
+            baseline.add(read.query_name)
         for read in fullref_bam.fetch(locus.chrom, start, end):
-            if not skip_read(read):
+            if not skip_read(read) and read.query_name in baseline:
```

Comment:
This is the main change so far. After looking at the results again, I'm not even sure how much of a difference this is making. The changes to the read numbers may have more to do with mapping stochastics.